modules/src/ics03_connection/error.rs

```diff
@@ -172,5 +172,8 @@ define_error! {
                 format_args!("the client state proof verification failed for client id {0}",
                     e.client_id)
             },
+
+        ImplementationSpecific
```

Comment:
Not sure if we use this anywhere.

Reply:
We don't use it in modules, but a user who implements a module handler could use it.
This error can be returned if another failure depending on the implementation happens.
For example, in the case where a connection end is stored in a key-value store, it fails due to the disk IO failure, and the caller of `ConnectionReader::connection_end(..)` can know that the failure isn't related to IBC and handle the error.

Reply:
Thank you for the clarification, that's very reasonable!